File: appendix02/course-tracker-building-webapp/course-tracker-final/src/main/java/com/manning/sbip/a02/service/CourseService.java
package com.manning.sbip.a02.service;

import com.manning.sbip.a02.model.Course;

import java.util.Optional;

public interface CourseService {

    Iterable<Course> createCourse(Course course);

    Optional<Course> findCourseById(int courseId);

    Iterable<Course> findAllCourses();

    Iterable<Course> updateCourse(Course course);

    Iterable<Course> deleteCourseById(int courseId);

}

File: ch10/Kotlin/course-tracker-kotlin-app-security/src/main/kotlin/com/manning/sbip/ch10/service/CourseService.kt
package com.manning.sbip.ch10.service

import com.manning.sbip.ch10.model.Course
import java.util.*

interface CourseService {
    fun createCourse(course: Course): Course
    fun findCourseById(courseId: Long): Course
    fun findAllCourses(): Iterable<Course>
    fun updateCourse(courseId: Long, updatedCourse: Course): Course
    fun deleteCourseById(courseId: Long)
}